Slim/Networking/Async.pm

```diff
@@ -164,14 +165,20 @@ sub _connect_error {
 
 sub _async_connect {
 	my ( $socket, $self, $args ) = @_;
-	
-	# Kill the timeout timer
-	Slim::Utils::Timers::killTimers( $socket, \&_connect_error );
-	
+
 	# check that we are actually connected
 	if ( !$socket->connected ) {
-		return _connect_error( $socket, $self, $args );
+		if ( ref $socket eq 'Slim::Networking::Async::Socket::HTTPS' and $! == EWOULDBLOCK ) {
+			# The TLS handshake is not yet complete.  Retry later.
+			return;
+		}
+		else {
+			return _connect_error( $socket, $self, $args );
```

Comment:
Wouldn't we need to kill the timer here?

Reply:
Conveniently, the first thing that `_connect_error()` does is kill its own timer.